nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -403,11 +423,215 @@ async fn run() -> Result<()> {
+    // have, if we see multiple answers with same counts from all task results
+    // we retry and wait for more results
+    // max_count count number of results having same max count
+    let mut max_count = 0;
+    // max_answer is number of results having max answer
+    let mut max_answer = 0;
+    let mut answer = String::new();
+    for (result, count) in answers {
+        if count == max_answer {
+            max_count += 1;
+        } else if count > max_answer {
+            max_count = 1;
+            max_answer = count;
+            answer = result;
+        }
+    }
+    // This is when we are going to retry because there is not a single
+    // answer occurs more than any others
+    let mut accepted_result: Option<String> = None;
+    if max_count == 1 {
+        accepted_result = Some(answer);
+    }
+    let (result, winners) = get_winners(tasks, accepted_result);
+    // TODO(wangjun.hong): Replace below with a single update
+
+    if !winners.is_empty() && !result.is_none() {
+        reward(assign_event_id.clone(), &winners, result.unwrap(), classifier_result_collection,
+        collection, 1).await;
+    }
+
+    Ok(())
+}
+
+/**
+Go through all tasks and return all winners and the result payload
+When result is None, no winners are returned
+**/
+fn get_winners(
+    tasks: &Vec<AssignerTask>,
+    result: Option<String>
+) -> (Option<ResultPayload>, Vec<Sender>) {
+    tracing::debug!("Result is: {:?}", result);
+    let mut winners: Vec<Sender> = vec![];
+    let mut result_payload: Option<ResultPayload> = None;
+    tasks.iter().for_each(|task| {
+        let task_result = get_task_identifier(task);
+        if task_result == result && !task_result.is_none() {
+            winners.push(task.worker.clone());
+            result_payload = Some(task.result.clone());
+        }
+    });
+
+    (result_payload, winners)
+}
+
+// Return values from a task identifier, in case of
+// failure happened, return None
+fn get_task_identifier(
+    task: &AssignerTask
+) -> Option<String> {
+    match get_result_identifier(task.result.output.clone()) {
+        Ok(Some(val)) => {
+            Some(val)
+        }
+        Ok(None) => {
+            None
+        }
+        Err(err) => {
+            tracing::debug!("Failed to parse identifier, error: {}", err);
+            None
+        }
+    }
+}
+
+async fn reward(
+    assign_event_id: String,
+    workers: &Vec<Sender>,
+    payload: ResultPayload,
+    classifier_result_collection: &Collection<ClassifierResult>,
+    collection: &Collection<FinishedJobs>,
+    job_type: u16,
+) -> bool {
+    tracing::debug!(
+        "All the results for {} are consistent, sending",
+        assign_event_id
+    );
+    let raw_data_id = payload.header.raw_data_id.clone();
+    let result_identifier = get_result_identifier(payload.output.clone());
+    match result_identifier   {
+        Err(_) =>{},
+        Ok(None)=>{},
```

Comment:
If it's `None`, are we OK with doing nothing here? Can this function really continue if there is no identifier?